src/stores/Binding/slices/ToggleDisable.ts

```diff
@@ -0,0 +1,113 @@
+import type { BindingState } from 'src/stores/Binding/types';
+import type { NamedSet } from 'zustand/middleware';
+
+import produce from 'immer';
+import { isBoolean } from 'lodash';
+
+import { hasLength } from 'src/utils/misc-utils';
+
+export type ToggleDisableUpdate = {
+    uuid: string;
+    bindingIndex: number;
+    val: boolean;
+};
+
+export interface StoreWithToggleDisable {
+    generateToggleDisableUpdates: (
+        targetUUIDs: string | string[] | null,
+        value?: boolean
+    ) => ToggleDisableUpdate[];
+    toggleDisable: (updates: ToggleDisableUpdate[]) => void;
+}
+
+export const getStoreWithToggleDisableSettings = (
+    set: NamedSet<StoreWithToggleDisable>
+): StoreWithToggleDisable => ({
+    toggleDisable: (updates) => {
+        set(
+            produce((state: BindingState) => {
+                updates.forEach(({ uuid, val }) => {
+                    const collectionName =
+                        state.resourceConfigs[uuid].meta.collectionName;
+
+                    if (val === true) {
```

Comment:
If `val` is a required boolean property of `ToggleDisableUpdate`, why do we need to explicitly check that it equates to `true`? I have seen this condition in a couple places in this PR.